stake-pool/program/src/processor.rs

```diff
@@ -1869,19 +1873,21 @@ impl Processor {
             .ok_or(StakePoolError::CalculationFailure)?;
 
         let new_pool_tokens = stake_pool
-            .calc_pool_tokens_for_deposit(all_deposit_lamports)
+            .calc_pool_tokens_for_deposit(stake_deposit_lamports + stake_rent)
```

Comment:
not technically needed, but just for paranoia
```suggestion
            .calc_pool_tokens_for_deposit(stake_deposit_lamports.saturating_add(stake_rent))
```